test/modules/Modules.Test/TestResultCoordinator/Reports/DirectMethod/DirectMethodReportGeneratorTest.cs

```diff
@@ -17,210 +18,40 @@ namespace Modules.Test.TestResultCoordinator.Reports.DirectMethod
-                        new DateTime(2020, 1, 1, 9, 10, 21, 10),
-                        new DateTime(2020, 1, 1, 9, 10, 22, 10),
-                        new DateTime(2020, 1, 1, 9, 10, 23, 10),
-                        new DateTime(2020, 1, 1, 9, 10, 24, 10),
-                        new DateTime(2020, 1, 1, 9, 10, 24, 15)
-                    },
-                    10, 6, 0, 0, 0, 0, 0, 1, 0, true
-                },
-                new object[]
-                {
-                    // MismatchFailure test
-                    Enumerable.Range(1, 6).Select(v => v.ToString()),
-                    new[] { "1", "2", "3", "4", "5", "6", "7" },
-                    new int[] { 200, 200, 200, 200, 200, 200, 200 },
-                    new DateTime[]
-                    {
-                        new DateTime(2020, 1, 1, 9, 10, 12, 10),
-                        new DateTime(2020, 1, 1, 9, 10, 13, 10),
-                        new DateTime(2020, 1, 1, 9, 10, 21, 10),
-                        new DateTime(2020, 1, 1, 9, 10, 22, 10),
-                        new DateTime(2020, 1, 1, 9, 10, 23, 10),
-                        new DateTime(2020, 1, 1, 9, 10, 24, 10),
-                        new DateTime(2020, 1, 1, 9, 10, 24, 15)
-                    },
-                    10, 6, 0, 0, 0, 0, 0, 0, 1, false
-                },
-                new object[]
-                {
-                    Enumerable.Range(1, 10).Select(v => v.ToString()),
-                    new[] { "1", "2", "3", "8", "10", "11" },
-                    new int[] { 200, 200, 200, 500, 500, 500, 500, 200, 200, 200 },
-                    new DateTime[]
-                    {
-                        // Smoke test for mixed results
-                        new DateTime(2020, 1, 1, 9, 10, 12, 10), // NetworkOnSuccess
-                        new DateTime(2020, 1, 1, 9, 10, 13, 10), // NetworkOnSuccess
-                        new DateTime(2020, 1, 1, 9, 10, 15, 11), // NetworkOffToleratedSuccess
-                        new DateTime(2020, 1, 1, 9, 10, 17, 10), // NetworkOffSuccess
-                        new DateTime(2020, 1, 1, 9, 10, 18, 10), // NetworkOffSuccess
-                        new DateTime(2020, 1, 1, 9, 10, 20, 12), // NetworkOnToleratedSuccess
-                        new DateTime(2020, 1, 1, 9, 10, 21, 15), // NetworkOnFailure
-                        new DateTime(2020, 1, 1, 9, 10, 24, 15),
-                        new DateTime(2020, 1, 1, 9, 10, 24, 17),
-                        new DateTime(2020, 1, 1, 9, 10, 25, 20) // NetworkOffFailure
-                        // Mismatch Success is the missing 9
-                        // MismatchFailure is the presence of 11 in the actualStoreValues
-                    },
-                    10, 3, 2, 1, 1, 1, 1, 1, 1, false
-                }
-            };
+        public static IEnumerable<object[]> GetCreateReportDataForTwoSources => DirectMethodReportDataWithSenderAndReceiverSource.GetCreateReportData;
```

Comment:
I think you can directly use DirectMethodReportDataWithSenderAndReceiverSource.GetCreateReportData in MemberData attribute.